crypto/ec/ec_local.h

```diff
@@ -323,7 +323,7 @@ struct ec_point_st {
 static ossl_inline int ec_point_is_compat(const EC_POINT *point,
                                           const EC_GROUP *group)
 {
-    return group->meth == point->meth
+    return group->meth->field_type == point->meth->field_type
```

Comment:
This is too permissive: both P256 and P384 (and any other prime curve) share the same field type, but can have very different ec_methods.

The real solution here would be to make sure the engine is loaded before the certificate is parsed and that it is the one creating the EVP_PKEY object inside the certificate when the certificate is loaded.

Reply:
About the field_type, indeed,  it's too permissive. But i don't know how to fix this problem. 

Sorry about the real solution you supplied, i'm not sure i understand what you mean.
Here is how i write the code in order:
1. tasshsmsm2_e = ENGINE_by_id();
2. ENGINE_init(tasshsmsm2_e)
3. ctx = SSL_CTX_new(meth);
4. SSL_CTX_use_certificate_file(ctx, SERVER_SIGN_CERT_HSM, SSL_FILETYPE_PEM);
5. EVP_PKEY *pkey = ENGINE_load_private_key();
6. SSL_CTX_use_PrivateKey(ctx, pkey);

In step 6, it doesn't work if i use the  libcrypto.a. Is there a problem with my order of calls?